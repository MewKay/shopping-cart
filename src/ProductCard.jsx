import PropTypes from "prop-types";

const ProductCard = ({ title }) => {
  return <li>{title}</li>;
};

ProductCard.propTypes = {
  title: PropTypes.string,
};

export default ProductCard;
