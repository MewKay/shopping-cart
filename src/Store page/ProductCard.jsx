import PropTypes from "prop-types";

const ProductCard = ({ imgURL, title, price, indexToShow, onClick }) => {
  return (
    <button onClick={() => onClick(indexToShow)}>
      <div className="product-image">
        <img src={imgURL} alt="" />
      </div>
      <p>{title}</p>
      <p>${price}</p>
    </button>
  );
};

ProductCard.propTypes = {
  imgURL: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  indexToShow: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ProductCard;
