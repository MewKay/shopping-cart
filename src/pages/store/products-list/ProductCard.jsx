import PropTypes from "prop-types";
import styles from "./ProductCard.module.css";

const ProductCard = ({ imgURL, title, price, indexToShow, onClick }) => {
  return (
    <button
      className={styles["container"]}
      onClick={() => onClick(indexToShow)}
    >
      <div className={styles["product-image"]}>
        <img src={imgURL} alt="" />
      </div>
      <p className={styles["item-title"]}>{title}</p>
      <p className={styles["item-price"]}>${price}</p>
    </button>
  );
};

ProductCard.propTypes = {
  imgURL: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  indexToShow: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ProductCard;
