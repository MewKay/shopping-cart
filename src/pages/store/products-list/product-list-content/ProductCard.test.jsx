import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import ProductCard from "./ProductCard";

//Component props
const imgURL = "https://image.placeholder.com/product";
const title = "Product 0";
const price = 50;
const indexToShow = 0;
const onClick = vi.fn();

describe("ProductCard component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ProductCard
        imgURL={imgURL}
        title={title}
        price={price}
        indexToShow={indexToShow}
        onClick={onClick}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("calls onClick function when clicked", async () => {
    const user = userEvent.setup();
    render(
      <ProductCard
        imgURL={imgURL}
        title={title}
        price={price}
        indexToShow={indexToShow}
        onClick={onClick}
      />
    );

    const productCard = screen.getByRole("button", { name: /Product 0/ });
    await user.click(productCard);

    expect(onClick).toHaveBeenCalledWith(indexToShow);
  });
});
