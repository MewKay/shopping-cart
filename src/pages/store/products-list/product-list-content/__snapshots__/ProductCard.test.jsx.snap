// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProductCard component > renders correctly 1`] = `
<div>
  <button
    class="_container_454344"
  >
    <div
      class="_product-image_454344"
    >
      <img
        alt=""
        src="https://image.placeholder.com/product"
      />
    </div>
    <p
      class="_item-title_454344"
    >
      Product 0
    </p>
    <p
      class="_item-price_454344"
    >
      $
      50
    </p>
  </button>
</div>
`;
